src/core/atomic.d

```diff
@@ -76,10 +126,10 @@ version( CoreDdoc )
      * Returns:
      *  The result of the operation.
      */
-    HeadUnshared!(T) atomicOp(string op, T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc @safe
+    TailShared!T atomicOp(string op, T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc @safe
```

Comment:
Can't do ops on classes, so this is more permissive than before.
Can you recheck that `TailShared!T` is always implicitly convertible to `shared T`?